packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -0,0 +1,251 @@
+import '@aws-cdk/assert/jest';
+import { arrayWith, objectLike } from '@aws-cdk/assert';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import { App, Duration, Stack } from '@aws-cdk/core';
+import * as synthetics from '../lib';
+
+let stack: Stack;
+beforeEach(() => {
+  stack = new Stack(new App(), 'canaries');
+});
+
+test('Create a basic canary', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {});
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'canariescanary8f7842',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Canary can have specified name', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    Code: {
+      Handler: 'index.handler',
+      Script: 'exports.handler = async () => {\nconsole.log(\'hello world\');\n};',
+    },
+    RuntimeVersion: 'syn-1.0',
+  });
```

Comment:
This validation is excessive. All you should be checking here is that the `canaryName` that was specified is indeed used.

Make sure your tests stick to what they claim to be checking... Say here, the `RuntimeVersion` changes (because whatever reason) - this test has no reason to be touched because it's not it's goal to validate this.